### ISSUE TITLE
App push to project fails

### ISSUE THREAD

**Issue by migbro** [rabix/composer#457]
Any attempt to upload an app to a cavatica project returns an error of `Invalid JSON input`, even if rabix and cwltool both report no formatting errors.
![image (2)](https://user-images.githubusercontent.com/10851708/90149571-a136c780-dd52-11ea-960c-e892236159ef.png)

**Comment by kaushik-work** [comment]
Hello @migbro . Are you using a build from master? Could you try out building from the `1.5-rc1` branch please? This may be related to #458

**Comment by migbro** [comment]
Hi @kaushik-work, I was using the latest release, v1.0.2 available from master. I tried building from the branch `1.5-rc1` , and after running into an error while building (missing node-sass) and resolving with `npm i`, it fails with:
```
ERROR in /Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss:71
throw new Error("Module build failed: \n        &:not(\".text-inherit\") {\n             ^\n      Invalid CSS after \"        &:not(\": expected selector, was '\".text-inherit\") {'\n      in /Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss (line 17, column 15)");
^

Error: Module build failed:
        &:not(".text-inherit") {
             ^
      Invalid CSS after "        &:not(": expected selector, was '".text-inherit") {'
      in /Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss (line 17, column 15)
    at Object.rZm9 (/Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss:71:7)
    at __webpack_require__ (/Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss:20:30)
    at /Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss:63:18
    at /Users/brownm28/Documents/git_repos/composer/src/app/layout/action-bar/action-bar.component.scss:66:10
    at Script.runInContext (vm.js:142:20)
    at Script.runInNewContext (vm.js:148:17)
    at Object.runInNewContext (vm.js:290:38)
    at WebpackResourceLoader._evaluate (/Users/brownm28/Documents/git_repos/composer/node_modules/@ngtools/webpack/src/resource_loader.js:110:40)
    at Compilation.<anonymous> (/Users/brownm28/Documents/git_repos/composer/node_modules/@ngtools/webpack/src/resource_loader.js:58:26)
    at Compilation.applyPluginsAsyncSeries (/Users/brownm28/Documents/git_repos/composer/node_modules/tapable/lib/Tapable.js:206:13)
```
My OS env is macOS 10.14.6.
Thanks!

**Comment by kaushik-work** [comment]
Hi @migbro are you using node8? The build and tests work fine on the CI, which is pinned to node 8 (as mentioned in the Readme). Thanks!

**Comment by migbro** [comment]
Hi @kaushik-work , when I run `node -v`, my version reports as `v13.11.0`. Since the build works well in CI, would it be possible to make that built Mac GUI app available? I don't relish the idea of installing an older version of node on my local environment...I'm not too familiar with node and unaware of virtual env capabilities with node.
Thanks again for looking into it!

**Comment by kaushik-work** [comment]
@migbro did this problem just start happening? I realize from your answers above that you have been using 1.0.2 which is stable (i.e. we did not change it). It might be that this is a problem on the Cavatica side, and not Composer.

I would suggest to keep using your existing Composer version (which is stable) and I will follow up with Cavatica folks to ask if any new deploys have been done that might cause this. 

Thanks for reporting and sorry for the trouble.

**Comment by kaushik-work** [comment]
@migbro another follow up question: is this happening for every CWL, or specific ones? Thanks!

**Comment by migbro** [comment]
Hi @kaushik-work , this was for every CWL, even ones that previously pushed just fine, reported by multiple team members. It started at the time that I had posted this issue. We have since employed workarounds, like using the python api, etc, but your comments are quite helpful! I will reach back to the Cavatica team to look into what had changed, as perhaps they thought I was referring to the rabix CLI runner and not the GUI.

**Comment by migbro** [comment]
@kaushik-work , if it helps, our tools and workflows have been written in primarily yaml format, which had worked fine until recently.

**Comment by kaushik-work** [comment]
@migbro thank you for your patience. 

It does sound like an issue on the Cavatica end, 
1. You did not change your RC version
2. CWL that worked previously started to fail to push
3. We've had internal users run into a similar issue.

Are you hand writing your CWL by any chance?

We have two tools specifically catering to that coding style:

1. Rabix/Benten language server that can be used with VS Code or VI. https://github.com/rabix/benten. It offers code intelligence, auto completion etc. for CWL
2. Rabix.sbpack that allows you to push your locally written CWL to any SB platform: https://github.com/rabix/sbpack

Both tools can be installed with `pipx` or `pip` (they are standard Python packages)

Thanks!

**Comment by kaushik-work** [comment]
@migbro thank you for your patience. Our team member @gungorbudak found a work around and we've included it in the latest release (1.5.0) which is now available on the releases page along with some other enhancements we've been planning. Thank you for reporting this issue.

If the current release does not fix this issue, please reopen and we will investigate further.

Closed by e02ab0d2ca2bdf69e2b7b2e4355d56b8552421f2
